runelite-client/src/main/java/net/runelite/client/ui/overlay/components/ImageComponent.java

```diff
@@ -52,16 +55,27 @@ public Dimension render(Graphics2D graphics)
 			return null;
 		}
 
-		graphics.drawImage(image, preferredLocation.x, preferredLocation.y, null);
+		BufferedImage toRender = scaledImage != null ? scaledImage : image;
+		graphics.drawImage(toRender, preferredLocation.x, preferredLocation.y, null);
 		final Dimension dimension = new Dimension(image.getWidth(), image.getHeight());
 		bounds.setLocation(preferredLocation);
 		bounds.setSize(dimension);
 		return dimension;
 	}
 
 	@Override
-	public void setPreferredSize(Dimension dimension)
+	public void setPreferredSize(Dimension preferredSize)
 	{
-		// Just use image dimensions for now
+		boolean nulled = preferredSize == null || image == null;
+		boolean sameSize = !nulled && (preferredSize.width == image.getWidth() && preferredSize.height == image.getHeight());
```

Comment:
By the time this is evaluated in the following conditional, `!nulled` will always be true.
```suggestion
		boolean sameSize = preferredSize.width == image.getWidth() && preferredSize.height == image.getHeight();
```

Reply:
Scratch that, this is to avoid NPE